cpp/src/Slice/Grammar.y

```diff
@@ -313,91 +307,58 @@ definitions
 // ----------------------------------------------------------------------
 definition
 // ----------------------------------------------------------------------
-: module_def
+: module_def opt_semicolon
 {
     assert($1 == nullptr || dynamic_pointer_cast<Module>($1));
 }
-opt_semicolon
```

Comment:
What we're doing here is called a 'midrule action'.
Where user code runs in the middle of an actions.

There is no benefit to checking between `module Foo` and the optional semicolon.
So I simplified this into a normal rule...